cmd/notify/notify.go

```diff
@@ -0,0 +1,180 @@
+// Notify service, for sending email notifications
+package main
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/smtp"
+	"os"
+	"sda-pipeline/internal/broker"
+	"sda-pipeline/internal/common"
+	"sda-pipeline/internal/config"
+	"strconv"
+
+	"github.com/rabbitmq/amqp091-go"
+	log "github.com/sirupsen/logrus"
+)
+
+const err = "error"
+const ready = "ready"
+
+func main() {
+	conf, err := config.NewConfig("notify")
+	if err != nil {
+		log.Fatal(err)
+	}
+	mq, err := broker.NewMQ(conf.Broker)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer mq.Channel.Close()
+	defer mq.Connection.Close()
+
+	go func() {
+		connError := mq.ConnectionWatcher()
+		log.Error(connError)
+		os.Exit(1)
+	}()
+
+	forever := make(chan bool)
+
+	log.Infof("Starting %s notify service", conf.Broker.Queue)
+
+	go func() {
+		messages, err := mq.GetMessages(conf.Broker.Queue)
+		if err != nil {
+			log.Fatalf("Failed to get message from mq (error: %v)", err)
+		}
+
+		for d := range messages {
+			log.Debugf("received a message: %s", d.Body)
+
+			if err := validator(conf.Broker.Queue, conf.Broker.SchemasPath, d); err != nil {
+				log.Errorf("Failed to handle message, reason: %v", err)
+
+				continue
+			}
+			user := getUser(conf.Broker.Queue, d.Body)
+			if user == "" {
+				log.Errorln("No user in message, skipping")
+
+				continue
+			}
+
+			if err := sendEmail(conf.Notify, "THIS SHOULD TAKE A TEMPLATE", user, setSubject(conf.Broker.Queue)); err != nil {
```

Comment:
`"THIS SHOULD TAKE A TEMPLATE"` we leave like this or we configure an external file to read from?
should/can we use go templates for this ?

or we solve this in a different PR where we can add multiple templates depending on what happened per queue ?

Reply:
The idea was to have a template file depending on  the reason for the email being sent, to be solved down the road in separate PRs.

Reply:
i am ok with this.